analysis_driver/util/__init__.py

```diff
@@ -45,3 +48,40 @@ def _write_bcbio_csv(run_dir, sample_id, fastqs, user_sample_id):
             writer.writerow([fq, user_sample_id])
 
     return csv_file
+
+
+def check_bcl_files(basecalls_dir, run_info, validation_log):
+    tile_ids = run_info.tiles
+    cycles = sum(int(e.attrib['NumCycles']) for e in run_info.mask.reads)
+    all_bcls = []
+
+    validated_bcls = []
+    if isfile(validation_log):
+        with open(validation_log, 'r') as f:
+            validated_bcls = [l.rstrip('\n') for l in f.readlines()]
+
+    for t in tile_ids:
+        lane = t[2]  # s_1_1101.bcl.gz
+        lane_dir = 'L00' + lane
+        all_bcls.extend(join(lane_dir, 'C' + str(c + 1) + '.1', t + '.bcl.gz') for c in range(cycles))
+
+    files_to_check = [f for f in all_bcls if f not in validated_bcls]
+
+    app_logger.info('Checking %s files', len(files_to_check))
+    e = ArrayExecutor(['gzip -t %s' % join(basecalls_dir, f) for f in files_to_check], stream=True)
```

Comment:
In production it would be staggered as the bcls are generated, but it could be a lot of parallel processes if for some reason we run it on a complete dataset. Will run it through executor.